configure.ac

```diff
@@ -28,6 +28,10 @@ AC_PROG_CPP
 AC_CONFIG_MACRO_DIR([m4])
 AX_CXX_COMPILE_STDCXX([14], [noext], [mandatory])
 
```

Comment:
If I understand this code, it will always try to append `-std=gnu11`, assuming success.  But if the default C compiler of the user is already C11-compatible, then this just adds confusion.  Shouldn't we be conditionally appending, instead?

And if I misunderstood this code, then let's add a comment.